test/dynamo/test_subclasses.py

```diff
@@ -37,13 +37,13 @@ def __call__(self, gm: torch.fx.GraphModule, example_inputs):
 
 @contextlib.contextmanager
 def preserve_subclass_config():
-    old_subclass_set = set(torch._dynamo.config.traceable_tensor_subclasses)
+    old_subclass_set = list(torch._dynamo.config.traceable_tensor_subclasses)
     try:
-        torch._dynamo.config.traceable_tensor_subclasses.add(MockSubclass)
+        torch._dynamo.config.traceable_tensor_subclasses.append(MockSubclass)
         yield
     finally:
         torch._dynamo.config.traceable_tensor_subclasses.clear()
-        torch._dynamo.config.traceable_tensor_subclasses.update(old_subclass_set)
+        torch._dynamo.config.traceable_tensor_subclasses = old_subclass_set
```

Comment:
Are you sure? update() would have ensured other aliases to the set got the update, but now you are creating a new mutable object